mongo/integration/load_balancer_prose_test.go

```diff
@@ -0,0 +1,101 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/internal/testutil/assert"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func TestLoadBalancerSupport(t *testing.T) {
+	mt := mtest.New(t, mtest.NewOptions().Topologies(mtest.LoadBalanced).CreateClient(false))
+	defer mt.Close()
+
+	mt.Run("RunCommandCursor pins to a connection", func(mt *mtest.T) {
+		// The LB spec tests cover the behavior for cursors created by CRUD operations, but RunCommandCursor is
+		// Go-specific so there is no spec test coverage for it.
+
+		initCollection(mt, mt.Coll)
+		findCmd := bson.D{
+			{"find", mt.Coll.Name()},
+			{"filter", bson.D{}},
+			{"batchSize", 2},
+		}
+		cursor, err := mt.DB.RunCommandCursor(mtest.Background, findCmd)
+		assert.Nil(mt, err, "RunCommandCursor error: %v", err)
+		defer func() {
+			_ = cursor.Close(mtest.Background)
+		}()
+
+		assert.True(mt, cursor.ID() > 0, "expected cursor ID to be non-zero")
+		assert.Equal(mt, 1, mt.NumberConnectionsCheckedOut(),
+			"expected one connection to be checked out, got %d", mt.NumberConnectionsCheckedOut())
+	})
+
+	mt.RunOpts("wait queue timeout errors include extra information", noClientOpts, func(mt *mtest.T) {
+		// There are spec tests to assert this behavior, but they rely on the waitQueueTimeoutMS Client option, which is
+		// not supported in Go, so we have to skip them. These prose tests make the same assertions, but use context
+		// deadlines to force wait queue timeout errors.
+
+		assertErrorHasInfo := func(mt *mtest.T, err error, numCursorConns, numTxnConns, numOtherConns int) {
+			mt.Helper()
+
+			assert.NotNil(mt, err, "expected wait queue timeout error, got nil")
+			expectedMsg := fmt.Sprintf("maxPoolSize: 1, "+
+				"connections in use by cursors: %d, "+
+				"connections in use by transactions: %d, "+
+				"connections in use by other operations: %d",
+				numCursorConns, numTxnConns, numOtherConns,
+			)
+			assert.True(mt, strings.Contains(err.Error(), expectedMsg),
+				"expected error %q to contain substring %q", err, expectedMsg)
+		}
+		maxPoolSizeMtOpts := mtest.NewOptions().
+			ClientOptions(options.Client().SetMaxPoolSize(1))
+
+		mt.RunOpts("cursors", maxPoolSizeMtOpts, func(mt *mtest.T) {
+			initCollection(mt, mt.Coll)
+			findOpts := options.Find().SetBatchSize(2)
+			cursor, err := mt.Coll.Find(mtest.Background, bson.M{}, findOpts)
+			assert.Nil(mt, err, "Find error: %v", err)
+			defer func() {
+				_ = cursor.Close(mtest.Background)
+			}()
+
+			ctx, cancel := context.WithTimeout(mtest.Background, 50*time.Millisecond)
```

Comment:
Minor, does the exact time matter? If not, might as well use 1ms. If we add more tests like this in the future 50ms could add up.

Reply:
It does not technically matter, but I gave it some extra time to avoid flaky failures. Changed to 5ms.